source/framework/core/inc/TRestEventProcess.h

```diff
@@ -102,75 +102,64 @@ class TRestEventProcess : public TRestMetadata {
         std::string type = REST_Reflection::GetTypeName<T>();
         return (T*)GetMetadata(type);
     }
-    TRestMetadata* GetMetadata(std::string nameortype);
-    TRestEventProcess* GetFriend(std::string nameortype);
-    TRestEventProcess* GetFriendLive(std::string nameortype);
-    int GetNumberOfParallelProcesses() { return fParallelProcesses.size(); }
+    TRestMetadata* GetMetadata(std::string nameOrType);
+    TRestEventProcess* GetFriend(std::string nameOrType);
+    TRestEventProcess* GetFriendLive(std::string nameOrType);
+    inline size_t GetNumberOfParallelProcesses() const { return fParallelProcesses.size(); }
     TRestEventProcess* GetParallel(int i);
     //////////////////////////////////////////////////////////////////////////
-    /// \brief Set observable value for analysistree.
+    /// \brief Set observable value for AnalysisTree.
     ///
     /// It will rename the observable to "processName_obsName"
     /// If use dynamic observable, it will try to create new observable
     /// in the AnalysisTree if the observable is not found
     template <class T>
-    void SetObservableValue(std::string name, const T& value) {
+    inline void SetObservableValue(std::string name, const T& value) {
```

Comment:
In general is not a good practise to have such a long `inline` functions

Reply:
> Looks fine to me although I left some comments. I think we should also change the inherited functions in libraries and perhaps packages. In that case, can you post the related PR in this PR?

Thanks for the comments, but the PR is still work in progress! I will link everything in this PR when I have finished.

Reply:
I agree, but I couldn't change the original implementation, perhaps you know how to do it? remember this is a template function, its not straightforward to define in the cxx.

Reply:
Indeed, I dind't realize that it is a template function. The implementation is relatively easy, however the explicit instantiation is a lot of headache. You can leave as it is.